src/main/java/io/vertx/cassandra/impl/ResultSetImpl.java

```diff
@@ -30,9 +30,8 @@
   public ResultSetImpl() {
   }
 
-  ResultSetImpl(com.datastax.driver.core.ResultSet resultSet) {
-    this.resultSet = resultSet;
-    rows = resultSet.all();
+  ResultSetImpl(List<Row> rows) {
```

Comment:
So now ResultSet would always be created with all rows loaded?

Reply:
Yes

Reply:
What's the rationale for this? Provide an API similar to reactive-pg (all=result set; 1 by 1=streaming; flexible=cursor?)

Reply:
Becouse for now the wrapper is just useless

Reply:
Yes, for now the wrapper would be useless. But what's the plan in the longer term? See my previous comment?

Reply:
So, you wonder how we will implement cursor then if we will return `List<Row>` here?

Reply:
And I actually think whether we need a meta information for the result.

Reply:
If we would use just  `List<Row>` here we will just lose it.

Reply:
@tsegismont , so your point is that we definitely should keep `ResultSet`, but enriching it with information received from original ResultSet, also having it as a constructor param.

Reply:
My point is that the driver's `ResultSet` object is very flexible with respect to how the data is loaded. You can get a few rows (`one()`), then fetch more (`fetchMoreResults()`), then all the remaining rows (`all`). With these changes, we would always load all rows upfront.

Reply:
Yes, I agree. But I still did not get what we should do here. So the `com.datastax.driver.core.ResultSet` should be only constructor param, and we should do fetching inside the `ResultSet` object?

Reply:
I think the Vert.x client `ResultSet` interface should look like:

```java
interface ResultSet {
  boolean isExhausted();
  boolean isFullyFetched();
  int getAvailableWithoutFetching();
  void fetchMoreResults(Handler<AsyncResult<Void>> handler);
  void one(Handler<AsyncResult<Row>> handler)
  void all(Handler<AsyncResult<List<Row>>> handler)
  ColumnDefinitions getColumnDefinitions();
  boolean wasApplied();
}
```

In other words, a non-blocking version and vertxy version of the driver's `ResultSet`

Reply:
Ah, I got what you mean. In fact, I am not sure about this, since the if the API will look in this way, it would be complex(not only about because having a lot of callbacks) to use, because in most cases user doesn't need to use such a low-level API.

Maybe along with the API we can provide a simplified version? So, i mean having:

```java
execute(String query, Handler<AsyncResult<List<Row>>> resultHandler);
```

along with 

```java
execute(String query, Handler<AsyncResult<List<ResultSet>>> resultHandler);
```

where `ResultSet` is the version of Vert.x client ResultSet interface you have just proposed.

Reply:
+1 to this shortcut:

```java
execute(String query, Handler<AsyncResult<List<Row>>> resultHandler);
```
For the second one, it should be:

```java
execute(String query, Handler<AsyncResult<ResultSet>> resultHandler);
```

Reply:
Perhaps if you later design something similar to the cursor API in reactive-pg client, `ResultSet` would not be needed at all. But then we have to add `getColumnDefinitions` and `wasApplied` there, because these have their use cases and cannot be missed.

Reply:
Got it, done